probe-rs/src/flashing/flash_algorithm.rs

```diff
@@ -216,12 +216,23 @@ impl FlashAlgorithm {
 
         let assembled_instructions = raw.instructions.chunks_exact(size_of::<u32>());
 
-        if !assembled_instructions.remainder().is_empty() {
-            return Err(FlashError::InvalidFlashAlgorithmLength {
-                name: raw.name.to_string(),
-                algorithm_source: Some(target.source().clone()),
-            });
-        }
+        let last_elem = if !assembled_instructions.remainder().is_empty() {
+            let remainder = assembled_instructions.remainder();
```

Comment:
This can go before the if and then used in the if condition, no?

Reply:
Changed it then